### ISSUE TITLE
Supabase start fails with unexpected EOF

### ISSUE THREAD

**Issue by tirthpatell** [mckaywrigley/chatbot-ui#1095]
Just as the title says, everything was going smoothly until it reached that particular sql migrastion file `20240108234545_add_file_items.sql`. And then it fails with unexpected EOF error

![CleanShot 2024-01-10 at 17 56 48@2x](https://github.com/mckaywrigley/chatbot-ui/assets/32465104/c5328d68-c45e-4198-8fc8-ff8d05836a96)

**Comment by JesseGuerrero** [comment]
I get a similar issue with the first .sql file
![image](https://github.com/mckaywrigley/chatbot-ui/assets/27308928/49f91eb9-0457-4b17-9d47-913984a48681)

Arch Linux...

**Comment by kkartunov** [comment]
The `vector` type and the `hnsw` index method are part of the `pgvector` extension. Make sure that this extension is installed and enabled in your PostgreSQL database.

I am not sure how to do that in supabase. Tried but faild. Thus for now just commenting the index creation lines will get you going. App will work without indexes on those fields, probably slower but for local is not so important.

![image](https://github.com/mckaywrigley/chatbot-ui/assets/5585002/1501c0de-85fc-4099-8470-5b5d2a74678c)

**Comment by tirthpatell** [comment]
Can confim, the solution by @kkartunov worked. Thanks. I am going to close this.

**Comment by spammenotinoz** [comment]
Hi, after much effort I was able to re-create the EOF issue, believe the cause was not having NodeJS at the required level prior to the install, and or the default node still being the older version.

#Cleanup (This should remove old versions of NodeJS, but only if installed via apt-get
sudo npm cache clean -f
sudo npm install -g n
sudo n stable
apt-get remove nodejs

#Install Node (This should install the latest and change the default to the new install, in-case the removal did not work)
nvm install node
nvm install --lts
/home/linuxbrew/.linuxbrew/bin/brew reinstall supabase/tap/supabase #Just to be safe, re-install supabase (can't run as root)
docker volume rm supabase_config_chatbotui supabase_db_chatbotui supabase_edge_runtime_chatbotui supabase_inbucket_chatbotui supabase_storage_chatbotui #deletes the old supabase docker volumes
rm -R (your chatbot-ui directory)  #important as the failed start modifes files, you want to start with a clean clone
#Validate the chatbot-ui folder is gone
edit your config.toml as required
edit your migration file as required
get back in the chatbot-ui folder and run
update your .env.local or set your system\variables, whatever approach you us
supabase start

**Comment by rdkdepak** [comment]
Hi, I’m encountering a similar issue with the first .sql file, but the error message is vague. I’ve tried a clean install of Node.js using version 18 as specified. I even commented out those index values, although it didn’t reach that part of the file. I also installed pgvector and PostgreSQL. However, I'm unsure how to integrate it with Supabase. I'm using an Ubuntu 22 VM on VirtualBox. I noticed a suggestion to set the CPU to "host," but that’s the default setting in VirtualBox.
![image](https://github.com/user-attachments/assets/cbe7324a-1beb-4d38-b4b8-3066edbd6de0)
